### ISSUE TITLE
Add ENS resolution to addresses

### ISSUE THREAD

**Issue by 0xTranqui** [1ifeworld/river#67]
Tldr we dont have ENS resolution in the app yet, and need to use a custom solution to facilitate this since our app doesnt live natively on L1 which would let it take advantage of built in wagmi support for ENS rendering (using the built in rendering uses rpc calls but thats a separate question).

I suspect that the cleanest way to do this will likely be via offchain indexer/resolver, such as this [hook](https://github.com/0xTranqui/seeknoevil/blob/master/src/hooks/useENSResolver.ts) I implemented in seeknoevil that used the basement indexer, but there are issues to this approach such as potentially incorrect resolution for smart contracts that dont necessarily have a matching address on L2 (or perhaps a diff address on L2)

Fortunately, it appears that "proper" L2 ens validation has been [implemented](https://github.com/ourzora/nouns-builder/pull/338) in nouns builder recently, which can probably take us the rest of the way to putting together a solution

**Comment by 0xTranqui** [comment]
what we should actually implement at some point:
- get the relevant chain for given address. ex: if we want the address of L1 creator, get L1. if we want the address of someone who added a listing to a channel on L2, get L2
- check to see if that address is a smart contract by calling "getCode" using viem. if interested in L1, then we can check for ens resolution. if interested in L2, do not resolve it, since it probably does not have a matching address on L2 that is linked to ENs
- if its not a smart contract, then its probably an EOA, and we are proabbly safe to resolve it based on L1

**Comment by salieflewis** [comment]
Resolved #198 

Didn't close because of the comment above, not sure if you want to port that into another issue or open up that discussion elsewhere.

cc: @0xTranqui

**Comment by salieflewis** [comment]
Gentle bump on this @0xTranqui

**Comment by 0xTranqui** [comment]
im closing this cuz our approach works and doing something like outlined heere ^ is much more of a force

**Comment by 0xTranqui** [comment]
will start a discussion tho
